ibm/service/power/resource_ibm_pi_snapshot.go

```diff
@@ -123,6 +135,12 @@ func resourceIBMPISnapshotCreate(ctx context.Context, d *schema.ResourceData, me
 		log.Printf("no volumeids provided. Will snapshot the entire instance")
 	}
 
+	if v, ok := d.GetOk(Arg_UserTags); ok {
+		if len(flex.FlattenSet(v.(*schema.Set))) > 0 {
```

Comment:
I guess we do not need to convert set twice here. Directly setting UserTags is fine even if it's empty.

Reply:
Fixed in latest commit